mlflow/projects/backend/local.py

```diff
@@ -131,12 +133,17 @@ def _invoke_mlflow_run_subprocess(
     return LocalSubmittedRun(run_id, mlflow_run_subprocess)
 
 
-def _build_mlflow_run_cmd(uri, entry_point, storage_dir, use_conda, run_id, parameters):
+def _build_mlflow_run_cmd(
+    uri, entry_point, docker_args, storage_dir, use_conda, run_id, parameters
+):
     """
     Build and return an array containing an ``mlflow run`` command that can be invoked to locally
     run the project at the specified URI.
     """
     mlflow_run_arr = ["mlflow", "run", uri, "-e", entry_point, "--run-id", run_id]
+    if docker_args is not None:
+        for key, value in docker_args.items():
+            mlflow_run_arr.extend(["--docker-args", "%s=%s" % (key, value)])
```

Comment:
@alfozan 
I think we should process `docker_args` the same way as `_get_docker_command`.

https://github.com/mlflow/mlflow/blob/1e88bac455e78fc212a252ff2fbab1267a62441f/mlflow/projects/backend/local.py#L197-L210

Reply:
We could extract the code above into a new function (e.g. `_process_docker_args`) and reuse it `_build_mlflow_run_cmd ` and `_get_docker_command`.

Reply:
> @alfozan
> I think we should process `docker_args` the same way as `_get_docker_command`.
> 

I looked into how `docker_args` is processes in function `_get_docker_command()` but that's for different purpose - albeit both are related to the same environment.

`_get_docker_command()` prepares `docker run ...` command, while this PR fixes `_build_mlflow_run_cmd()`, which generates the `mlflow run ...` command that is run in a subprocesses.

Further, 
before fix, the internally generated mlrun run cmd is
```
['mlflow', 'run', '/home/alfozan/mlflow/examples/docker', '-e', 'main', '--run-id', 'ce179496f2ff47c5b010900928f55401', '-P', 'alpha=0.5']
```
after fix:
```
['mlflow', 'run', '/home/alfozan/mlflow/examples/docker', '-e', 'main', '--run-id', 'ce179496f2ff47c5b010900928f55401', '--docker-args', 'name=mlflow-container-run', '-P', 'alpha=0.5']
```

Reply:
Sorry, you're right. so we need to convert `docker_args` (dict) into the format that `mlflow run` can accept (like `'--docker-args', 'name=mlflow-container-run'`).

Reply:
> So we need to convert `docker_args`...

I don't see the need to convert anything as it's already in the right format. Could you please elaborate?

Reply:
```suggestion
    if docker_args is not None:
        for key, value in docker_args.items():
            args = key if isinstance(value, bool) else "%s=%s" % (key, value)
            mlflow_run_arr.extend(["--docker-args", arg])
```

Reply:
> to allow flag parameters (such as `t`):

I see. current fix works even with flag params. Though the suggestion does not change the logic, it makes the internally generated mlrun run cmd look nicer so I merged it. thanks


```python
import mlflow

mlflow.projects.run(
    uri='examples/docker',
    parameters={"alpha": 0.5},
    synchronous=False,
    docker_args={"name": "mlflow-container-run", "tty": True},
)
```
generated mlrun run cmd:
```
['mlflow', 'run', '/home/alfozan/mlflow/examples/docker', '-e', 'main', '--run-id', 'd92d649bd6a84b6fb72ed3c77f2f4489', '--docker-args', 'name=mlflow-container-run', '--docker-args', 'tty', '-P', 'alpha=0.5']
```